mantle/cosa/schema.go

```diff
@@ -0,0 +1,74 @@
+package cosa
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+
+	schema "github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	// JSON Schema document. Default the generated Schema.
+	SchemaJSON = generatedSchemaJSON
+)
+
+func init() {
+	runtimeSchemaPath := os.Getenv("COSA_META_SCHEMA")
+	if runtimeSchemaPath != "" {
+		f, err := os.Open(runtimeSchemaPath)
+		if err != nil {
+			panic(errors.Wrapf(err, "failed to open schema file %s", runtimeSchemaPath))
+		}
+		defer f.Close()
+
+		if err := SetSchemaFromFile(f); err != nil {
+			panic(errors.Wrapf(err, "failed to read in schema file %s", runtimeSchemaPath))
+		}
+	}
+}
+
+// SetSchemaFromFile sets the validation JSON Schema
+func SetSchemaFromFile(r io.Reader) error {
+	if r == nil {
+		return errors.New("schema input is invalid")
+	}
+	in, err := ioutil.ReadAll(r)
+	if err != nil {
+		SchemaJSON = string(in)
```

Comment:
This looks wrong, we're setting `SchemaJSON` if `err != nil` and ignoring `err`?

Reply:
Indeed, your right. Fixed and added a test.

Reply:
But we're still ignoring `err` right?  Any reason not to use the usual USB foot pedal every Golang programmer using the language for more than an hour has set up to insert `if err != nil { return err }` ?

Reply:
Indeed...dusting off my Golang after a little hiatus :) Re-looking at it was a bit wonky. Reworked and pushed.